instrumentation-core/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/httpclients/resttemplate/RestTemplateAutoConfiguration.java

```diff
@@ -43,6 +43,6 @@
   @Bean
   @Autowired
   public RestTemplateBeanPostProcessor otelRestTemplateBeanPostProcessor(final Tracer tracer) {
-    return new RestTemplateBeanPostProcessor(tracer);
+    return new RestTemplateBeanPostProcessor();
```

Comment:
in this case we want to pass `tracer` on, so that spring boot starter users can supply their own Tracer
```suggestion
    return new RestTemplateBeanPostProcessor(tracer);
```